tests/foreman/cli/test_discoveryrule.py

```diff
@@ -110,6 +110,7 @@ def test_positive_create_with_name(self, name, discoveryrule_factory):
         """
         rule = discoveryrule_factory(options={'name': name})
         assert rule.name == name
+        DiscoveryRule.delete({'id': rule.id})
```

Comment:
Just a suggestion to handle the teardown (of deleting Rule) within try catch so that it wont impact in any case.